flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/polardbx/PolardbxSourceTCase.java

```diff
@@ -0,0 +1,472 @@
+                                                        return m.matches() ? m.group(1) : x;
+                                                    })
+                                            .collect(Collectors.joining("\n"))
+                                            .split(";"))
+                            .collect(Collectors.toList());
+            for (String stmt : statements) {
+                statement.execute(stmt);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void testSingleKey() throws Exception {
+        int parallelism = 1;
+        String[] captureCustomerTables = new String[] {"orders"};
+
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+        env.setParallelism(parallelism);
+        env.enableCheckpointing(200L);
+        env.setRestartStrategy(RestartStrategies.fixedDelayRestart(1, 0));
+        String sourceDDL =
+                format(
+                        "CREATE TABLE orders_source ("
+                                + " id BIGINT NOT NULL,"
+                                + " seller_id STRING,"
+                                + " order_id STRING,"
+                                + " buyer_id STRING,"
+                                + " create_time TIMESTAMP,"
+                                + " primary key (id) not enforced"
+                                + ") WITH ("
+                                + " 'connector' = 'mysql-cdc',"
+                                + " 'scan.incremental.snapshot.enabled' = 'true',"
+                                + " 'hostname' = '%s',"
+                                + " 'port' = '%s',"
+                                + " 'username' = '%s',"
+                                + " 'password' = '%s',"
+                                + " 'database-name' = '%s',"
+                                + " 'table-name' = '%s',"
+                                + " 'scan.incremental.snapshot.chunk.size' = '100',"
+                                + " 'server-id' = '%s'"
+                                + ")",
+                        HOST_NAME,
+                        PORT,
+                        USER_NAME,
+                        PASSWORD,
+                        DATABASE,
+                        getTableNameRegex(captureCustomerTables),
+                        getServerId());
+
+        // first step: check the snapshot data
+        String[] snapshotForSingleTable =
+                new String[] {
+                    "+I[4, 1002, 2, 106, 2022-01-16T00:00]",
+                    "+I[5, 1003, 1, 107, 2022-01-16T00:00]",
+                    "+I[2, 1002, 2, 105, 2022-01-16T00:00]",
+                    "+I[3, 1004, 3, 109, 2022-01-16T00:00]",
+                    "+I[1, 1001, 1, 102, 2022-01-16T00:00]",
```

Comment:
minor: keep the record in id order as we use `assertEqualsInAnyOrder` next

Reply:
the order has been fixed